### PR TITLE
chore: port encoder to KLR

### PR DESCRIPTION
**Pull request by govereau**
The KLR language is the official intermediate format. This patch ports the serialization and deserialization code to this new data type.

### CODE REVIEW

File: NKL/Encode.lean [hunk 1]
```diff
@@ -151,117 +150,202 @@ private def decList (f : DecodeM a) : DecodeM (List a) := do
...
   | t => throw s!"Unknown tag in Expr {t}"
+where
+  decKeyword : DecodeM (String × Expr) :=
+    return ((<- decString), (<- decExpr))
 
 private def chkExpr (e : Expr) : Bool :=
   (decode' decExpr $ encExpr e) == some e
 
-private def nil := Expr.value .nil
+private def nil := Expr.const .none
```
Comment (seanmcl) [inline NKL/Encode.lean:302]: Call this `none`?

File: NKL/Encode.lean [hunk 2]
```diff
@@ -151,117 +150,202 @@ private def decList (f : DecodeM a) : DecodeM (List a) := do
...
-  | .bvar s           => tag 0x11 [encString s]
-  | .var s _          => tag 0x12 [encString s]
-  | .subscript e ix   => tag 0x13 [encExpr e, encList encExpr ix]
-  | .slice l u step   => tag 0x14 [encExpr l, encExpr u, encExpr step]
-  | .binop op l r     => tag 0x15 [encString op, encExpr l, encExpr r]
-  | .cond c t e       => tag 0x16 [encExpr c, encExpr t, encExpr e]
-  | .tuple es         => tag 0x17 [encList encExpr es]
-  | .list es          => tag 0x18 [encList encExpr es]
-  | .call f ax        => tag 0x19 [encExpr f, encList encExpr ax]
-  | .gridcall f ix ax => tag 0x1a [encExpr f, encList encExpr ix, encList encExpr ax]
```
Comment (seanmcl) [inline NKL/Encode.lean:205]: what happened to gridcall?

File: NKL/Encode.lean (hunk 1 above)
Comment (govereau) [reply NKL/Encode.lean:302]: I was trying to avoid clashing with Lean's none.

File: NKL/Encode.lean (hunk 2 above)
Comment (govereau) [reply NKL/Encode.lean:205]: Likely it will have to be added back, depending on what we do in the tracing.

**Review by seanmcl — APPROVED** [review]

**Pull request merged**